developer/src/kmc-package/src/compiler/package-version-validation.ts

```diff
@@ -0,0 +1,139 @@
+import { KmpJsonFile, CompilerCallbacks, KpsFile, KmxFileReader, KMX } from '@keymanapp/common-types';
+import { CompilerMessages } from './messages.js';
+
+export class PackageVersionValidation {
+
+  constructor(private callbacks: CompilerCallbacks) {}
+
+  /**
+   * Verifies version information in corresponding keyboards and updates kmpJson
+   * metadata as the version information can be out of sync in the .kps file
+   * after update a contained keyboard.
+   * @param kpsFilename
+   * @param kps
+   * @param kmp
+   * @returns
+   */
+  public validateAndUpdateVersions(kpsFilename: string, kps: KpsFile.KpsFile, kmp: KmpJsonFile.KmpJsonFile) {
+    const followKeyboardVersion = kps.options?.followKeyboardVersion !== undefined;
+
+    if(followKeyboardVersion) {
+      if(!this.checkFollowKeyboardVersion(kps, kmp)) {
+        return false;
+      }
+    }
+
+    let result = true;
+
+    if(!kmp.keyboards) {
+      // Lexical model packages don't have version metadata
```

Comment:
... that doesn't sound right to me.

The lexical models don't track their version independently - it's only on the package.

So...

```suggestion
      // Lexical models don't have version metadata; only their packages.
```

-----

Just to confirm, I went and double-checked the `kmp.json` for the `nrc.en.mtnt` (default English) lexical model, and there's definitely version information in there.  (Using whatever compiler the lexical-models repo currently uses... which is probably 16.0's.)

```json
{
  "system": {
    "keymanDeveloperVersion": "14.0.180.0",
    "fileVersion": "12.0"
  },
  "options": {},
  "info": {
    "author": {
      "description": "Eddie Antonio Santos",
      "url": "mailto:easantos@ualberta.ca"
    },
    "copyright": {
      "description": "© 2019-2022 National Research Council Canada"
    },
    "name": {
      "description": "English language model mined from MTNT"
    },
    "version": {
      "description": "0.1.9"
    }
  },
  // ...
}
```

Now, maybe there's a difference in what sort of version info is available, but that's different from "no version metadata".

Of note, though:

```json
{
  // ...
  "files": [
    {
      "name": "nrc.en.mtnt.model.js",
      "description": "Lexical model nrc.en.mtnt.model.js",
      "copyLocation": 0
    }
  ],
  "lexicalModels": [
    {
      "name": "English dictionary (MTNT)",
      "id": "nrc.en.mtnt",
      "languages": [
        {
          "name": "English",
          "id": "en"
        },
        {
          "name": "English (US)",
          "id": "en-us"
        },
        {
          "name": "English (Canada)",
          "id": "en-ca"
        }
      ]
    }
  ],
  "startMenu": {}
}
```

No version metadata on the _model_.